pkg/dashboard/ui/src/app/shared/services/import.service.js

```diff
@@ -74,28 +96,109 @@
                     }
                 })
                 .then(function () {
-                    return NuclioProjectsDataService.getProjects();
+                    return NuclioProjectsDataService.getProjects()
                 })
                 .then(function () {
                     var projectName = lodash.get(project, 'metadata.name');
                     var functions = lodash.get(project, 'spec.functions');
 
-                    lodash.forEach(functions, function (func) {
-                        NuclioFunctionsDataService.createFunction(func, projectName);
+                    var checkedFunctionList = lodash.map(functions, function (func) {
+                        return NuclioFunctionsDataService.createFunction(func, projectName)
+                            .catch(function (error) {
+                                if (error.status === 409) {
+                                    if (lodash.has(conflictProjectsData, projectName)) {
+                                        return conflictProjectsData[projectName].push(func);
+                                    } else {
+                                        return lodash.set(conflictProjectsData, [projectName], [func]);
+                                    }
```

Comment:
Please note that `Array.prototype.push` method returns the new pushed item:
```js
var arr = [1, 2, 3];
arr.push(4)
// => 4
```
On the other hand `lodash.set` returns the object you provide it as the first argument in the function call:
```js
var obj = {};
lodash.set(obj, 'key', 'value');
// => { key: 'value' }
```
So you're not retuning the same kind of thing in the different `if/else` branches.
Maybe you'd want to change the `if` branch to:
```js
if (lodash.has(conflictProjectsData, projectName)) {
    conflictProjectsData[projectName].push(func);
    return conflictProjectsData[projectName];
}
```